homeassistant/components/wled/__init__.py

```diff
@@ -168,9 +164,9 @@ class WLEDDeviceEntity(WLEDEntity):
     def device_info(self) -> DeviceInfo:
         """Return device information about this WLED device."""
         return {
-            ATTR_IDENTIFIERS: {(DOMAIN, self.coordinator.data.info.mac_address)},
-            ATTR_NAME: self.coordinator.data.info.name,
-            ATTR_MANUFACTURER: self.coordinator.data.info.brand,
-            ATTR_MODEL: self.coordinator.data.info.product,
-            ATTR_SOFTWARE_VERSION: self.coordinator.data.info.version,
+            "identifiers": {(DOMAIN, self.coordinator.data.info.mac_address)},
+            "name": self.coordinator.data.info.name,
+            "manufacturer": self.coordinator.data.info.brand,
+            "model": self.coordinator.data.info.product,
+            "sw_version": self.coordinator.data.info.version,
```

Comment:
I'm pretty sure this is a bug? I'm not see why else this change is needed.

Reply:
@frenck `DeviceInfo` is now `TypedDict` which verifies key names. But it's still possible to use constants here if they're marked with `Final`.

Reply:
Right, but constants should be final... right? So instead of removing all constants everywhere, we should maybe fix the issue we have.

Reply:
I mean with `TypedDict` constants are not that useful anymore. I think their main purpose is prevent people from typos when they write the same string again and again. With `TypedDict` mypy already does that, so constant just duplicate this functionality. For example with constants adding new key requires adding it in 2 places rather than 1.

Reply:
> I mean with TypedDict constants are not that useful anymore.

Constants was never a typing protection, so I'm not sure what has changed at this point? I think it helps with things like auto-completion and finding uses of certain keys across the codebase.

Implementation: #50013